### PR TITLE
Add support for Zest scripts written in YAML

### PR DESCRIPTION
**Pull request by ricekot**

### CODE REVIEW

**Comment by thc202** [comment]
I was expecting a flag to keep the original format, guess that can be added later too.

File: build.gradle [hunk 1]
```diff
@@ -30,7 +30,10 @@ dependencies {
              'com.google.code.gson:gson:2.8.5',
```
Comment (thc202) [inline build.gradle:30]: We should look at removing this one and use a single library.

File: src/main/java/org/zaproxy/zest/core/v1/ZestYaml.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+package org.zaproxy.zest.core.v1;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+
+public class ZestYaml {
```
Comment (thc202) [inline src/main/java/org/zaproxy/zest/core/v1/ZestYaml.java:12]: Worth adding tests.

**Comment by psiinon** [comment]
It would be really good to see what the YAML looks like.
If its no documented then no one will actually use it, but this is a great start!

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**